soroban-env-host/src/vm.rs

```diff
@@ -281,85 +276,38 @@ impl Vm {
         func_sym: &Symbol,
         args: &[Val],
     ) -> Result<Val, HostError> {
-        host.charge_budget(ContractCostType::InvokeVmFunction, None)?;
-        host.with_frame(
-            Frame::ContractVM(self.clone(), *func_sym, args.to_vec()),
-            || {
-                let wasm_args: Vec<Value> = args
-                    .iter()
-                    .map(|i| Value::I64(i.get_payload() as i64))
-                    .collect();
-                let func_ss: SymbolStr = func_sym.try_into_val(host)?;
-                let ext = match self
-                    .instance
-                    .get_export(&*self.store.borrow(), func_ss.as_ref())
-                {
-                    None => {
-                        return Err(host.err(
-                            ScErrorType::WasmVm,
-                            ScErrorCode::MissingValue,
-                            "invoking unknown export",
-                            &[func_sym.to_val()],
-                        ))
-                    }
-                    Some(e) => e,
-                };
-                let func = match ext.into_func() {
-                    None => {
-                        return Err(host.err(
-                            ScErrorType::WasmVm,
-                            ScErrorCode::UnexpectedType,
-                            "export is not a function",
-                            &[func_sym.to_val()],
-                        ))
-                    }
-                    Some(e) => e,
-                };
-
-                self.wrapped_func_call(host, func_sym, &func, wasm_args.as_slice())
-            },
-        )
-    }
-
-    /// Invokes a function in the VM's module, converting externally stable XDR
-    /// [ScVal] arguments into [Host]-specific [Val]s and converting the
-    /// [Val] returned from the invocation back to an [ScVal].
-    ///
-    /// This function, like [Vm::new], is called as part of
-    /// [Host::invoke_function], and does not usually need to be called manually
-    /// from outside the crate.
-    //
-    // NB: This function has to take self by [Rc] because it stores self in
-    // a new Frame
-    #[cfg(test)]
-    pub(crate) fn invoke_function(
-        self: &Rc<Self>,
-        host: &Host,
-        func: &str,
-        args: &ScVec,
-    ) -> Result<ScVal, HostError> {
-        let func_sym = Symbol::try_from_val(host, &func)?;
-        let mut raw_args: Vec<Val> = Vec::new();
-        for scv in args.0.iter() {
-            raw_args.push(host.to_host_val(scv)?);
-        }
-        let raw_res = self.invoke_function_raw(host, &func_sym, raw_args.as_slice())?;
-        host.from_host_val(raw_res)
-    }
-
-    /// Returns a list of functions in the WASM module loaded into the [Vm].
-    pub fn functions(&self) -> Vec<VmFunction> {
```

Comment:
Why did you remove this function? 
I don't know if it's being used but theoretically it could be used by e.g. the sdk/CLI for displaying some useful info.

Reply:
Nobody seems to be using it. SDK and CLI use the spec to generate typed function specs (this will just return functions that operate on `Val`s, which isn't super useful). Also, I think we shouldn't expose any vm functions - if we ever need to get info for it, we'd rather do that via host interface (VM is too tightly coupled to the host now, so using it outside of the host leads to footguns, like the `invoke_function` issue I've encountered after refactoring.

Reply:
Sounds good. I'm not objecting to removing. Was just suggesting to double check.